user_profile/views.py

```diff
@@ -0,0 +1,242 @@
+    Login is required henceLoginRequiredMixin
+    """
+    template_name = 'pages/dashboard.html'
+    login_url = 'user_profile:index'
+    
+    def get_context_data(self, **kwargs):
+        """
+        Overridden get_context_data method
+        to include the original context data
+        as well as the user feed objects
+        which will be displayed
+        P.S. Can be better implemented using a ListView
+        rather than a templateView
+        """
+        context = super().get_context_data(**kwargs)
+        context['feed_objects'] = models.Feed.objects.filter(user_id=self.request.user.id)
+        return context
+
+
+
+class UpdateProfilePictureView(LoginRequiredMixin,View):
+    """
+    Update ProfilePicture View 
+    for updating user profile picture
+    accepts only POST requests
+    uses ProfilePictureForm that handles the update
+    """
+    login_url = 'user_profile:index'
+
+    def post(self,request,*args,**kwargs):
+        profile = request.user.profile
+        form = forms.ProfilePictureForm(request.POST,request.FILES,instance=profile)
+        if form.is_valid():
+            form.save()
+            messages.success(request,'Profile Picture Updated!')
+        else:
+            messages.error(request,"Something went wrong!")
+        return redirect('user_profile:settings')
+
+
+class UpdateCoverPictureView(LoginRequiredMixin,View):
+    """
+    Update CoverPicture View 
+    for updating user Cover picture
+    accepts only POST requests
+    uses ProfileCoverForm that handles the update
+    """
+    login_url = 'user_profile:index'
+
+    def post(self,request,*args,**kwargs):
+        profile = request.user.profile
+        form = forms.CoverPictureForm(request.POST,request.FILES,instance=profile)
+        if form.is_valid():
+            form.save()
+            messages.success(request,'Cover Picture Updated!')
+        else:
+            messages.error(request,"Something went wrong! ")
+        return redirect('user_profile:settings')
+
+class UpdateProfileView(LoginRequiredMixin,View):
+    """
+    Update user Profile View 
+    for updating user profile 
+    (proerties such as bio, education etc)
+    accepts only POST requests
+    uses ProfileUpdateform that handles the update
+    """
+    login_url = 'user_profile:index'
+
+    def post(self,request,*args,**kwargs):
+        profile = request.user.profile
+        form = forms.ProfileUpdate(request.POST,instance=profile)
+        if form.is_valid():
+            form.save()
+            request.user.first_name = form.cleaned_data['first_name'] if form.cleaned_data['first_name'] != request.user.first_name else request.user.first_name
+            request.user.last_name = form.cleaned_data['last_name'] if form.cleaned_data['last_name'] != request.user.last_name else request.user.last_name
+            request.user.email = form.cleaned_data['email'] if form.cleaned_data['email'] != request.user.email else request.user.email
```

Comment:
Do we need to add id else statements in these lines of code?

Reply:
they were needed for a previous logic i fixed it and have removed those